pandapower/toolbox/comparison.py

```diff
@@ -35,29 +36,59 @@ def dataframes_equal(df1, df2, ignore_index_order=True, **kwargs):
         df1 = df1.sort_index().sort_index(axis=1)
         df2 = df2.sort_index().sort_index(axis=1)
 
+    # geo columns are compared later
+    df1_cols = df1.columns.difference({"geo"})
+    df2_cols = df2.columns.difference({"geo"})
+
     # --- pandas implementation
     try:
-        pdt.assert_frame_equal(df1, df2, **kwargs)
-        return True
+        pdt.assert_frame_equal(df1[df1_cols], df2[df2_cols], **kwargs)
     except AssertionError:
         return False
 
-    # --- alternative (old) implementation
-    # if df1.shape == df2.shape:
-    #     if df1.shape[0]:
-    #         # we use numpy.allclose to grant a tolerance on numerical values
-    #         numerical_equal = np.allclose(df1.select_dtypes(include=[np.number]),
-    #                                       df2.select_dtypes(include=[np.number]),
-    #                                       atol=tol, equal_nan=True)
-    #     else:
-    #         numerical_equal = True
-    #     # ... use pandas .equals for the rest, which also evaluates NaNs to be equal
-    #     rest_equal = df1.select_dtypes(exclude=[np.number]).equals(
-    #         df2.select_dtypes(exclude=[np.number]))
-
-    #     return numerical_equal & rest_equal
-    # else:
-    #     return False
+    # --- compare geo columns
```

Comment:
Why this change? Could geo column not just be evaluated with everthing else? It is just a column of string with a json object that should be sorted by its keys alphabetically. So there should be no issues just letting it be evaluated by assert_frame_equal.

Reply:
if it is as you say, then I agree. But is it guaranteed that the keys are sorted alphabetically? Or is it as it is with dictionary keys?

Reply:
JSON and Python Dict do not enforce sorting. I should have used sorted keys everywhere, but might have missed some places. If you think it is better to not enforce key sorting that is ok, I just did not understand it from the code alone.

Reply:
I think, we can enforce sorting in all our codes but some users may not do so since it is ok for json, file i/o might also be a reason for not sorted keys.
For all those cases, I prefer a more general implementation in dataframes_equal().